gadget/mountedfilesystem.go

```diff
@@ -299,6 +307,43 @@ func fsStructBackupPath(backupDir string, ps *LaidOutStructure) string {
 	return filepath.Join(backupDir, fmt.Sprintf("struct-%v", ps.Index))
 }
 
+func maybePreserveManagedBootAssets(mountPoint string, ps *LaidOutStructure) ([]string, error) {
+	if ps.Role != SystemSeed && ps.Role != SystemBoot {
+		return nil, nil
+	}
+	// the assets are within the system-boot or system-seed partition, set
+	// the right flags so that files are looked for using their paths inside
+	// the partition
+	opts := &bootloader.Options{
+		Recovery:    ps.Role == SystemSeed,
+		NoSlashBoot: ps.Role == SystemBoot,
+	}
+	bl, err := bootloader.Find(mountPoint, opts)
+	if err != nil {
+		if err == bootloader.ErrBootloader {
+			// no bootloader in the partition?
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	mbl, ok := bl.(bootloader.ManagedAssetsBootloader)
+	if !ok {
+		// bootloader implementation does not support managing its
+		// assets
+		return nil, nil
+	}
+	managed, err := mbl.IsCurrentlyManaged()
```

Comment:
when we install uc20 the first time what happens here, we get false and then we overwrite these later?

Reply:
In install mode we'll write out the boot config. Then on first boot in run mode, the update-gadget-assets task is a nop until seeding is complete.

Reply:
As discussed on IRC, I'll add a todo in the writer code and a note in #8930 that the InstallBootConfig must come last.